### ISSUE TITLE
Parent constructor is protected, overriding constructor get a "possible useless method overriding detected" warning

### ISSUE THREAD

**Issue by weiwanning** [squizlabs/PHP_CodeSniffer#880]
`https://github.com/squizlabs/PHP_CodeSniffer/blob/master/CodeSniffer/Standards/Generic/Sniffs/CodeAnalysis/UselessOverridingMethodSniff.php#L17-L29`

Shouldn't we consider about if 
```php
class Foo
{
  protected function __construct($a, $b) {
    // code
  }
}

class Bar extends Foo
{
  public function __construct($a, $b) {
    parent::__construct($a, $b);
  }
}
```

Check for `Bar`'s constructor should not throw a Warning?

**Comment by aik099** [comment]
The PHP_CodeSniffer can't know the visibility of method in parent class, because it doesn't scan classes to see who extends what. That's why it's impossible to do what you ask.

**Comment by gsherwood** [comment]
What @aik099 said. There isn't anything I can do in PHPCS itself, so I recommend not using this sniff in your case, or just ignoring warnings it produces. They are just warnings, which basically mean "a code review may be required here".

Sorry I can't actually help resolve the issue.
